feature/experimental/bgp/ate_tests/bgp_remove_private_as/bgp_remove_private_as_test.go

```diff
@@ -0,0 +1,347 @@
+	if gotInstalled := gnmi.Get(t, dut, prefixesv4.Installed().State()); gotInstalled != wantInstalled {
+		t.Errorf("Installed prefixes mismatch: got %v, want %v", gotInstalled, wantInstalled)
+	}
+	if gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State()); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+// configureATE configures the interfaces and BGP protocols on an ATE, including
+// advertising some(faked) networks over BGP.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice, asSeg []uint32, asSEQMode bool) *ondatra.ATETopology {
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+
+	port2 := ate.Port(t, "port2")
+	iDut2 := topo.AddInterface(ateDst.Name).WithPort(port2)
+	iDut2.IPv4().WithAddress(ateDst.IPv4CIDR()).WithDefaultGateway(dutDst.IPv4)
+
+	// Setup ATE BGP route v4 advertisement.
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutSrc.IPv4).WithLocalASN(ateAS1).
+		WithTypeExternal()
+
+	bgpDut2 := iDut2.BGP()
+	bgpDut2.AddPeer().WithPeerAddress(dutDst.IPv4).WithLocalASN(ateAS2).
+		WithTypeExternal()
+
+	bgpNeti1 := iDut1.AddNetwork("bgpNeti1")
+	bgpNeti1.IPv4().WithAddress(advertisedRoutesv4CIDR).WithCount(routeCount)
+	bgpNeti1.BGP().WithNextHopAddress(ateSrc.IPv4)
+
+	if asSEQMode {
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSEQ()
+	} else {
+		// TODO : SET mode is not working
+		// https://github.com/openconfig/featureprofiles/issues/1659
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSET()
+	}
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	return topo
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+// verifyBGPAsPath is to Validate AS Path attribute using bgp rib telemetry on ATE.
+func verifyBGPAsPath(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice, asSeg []uint32, removeASPath bool) {
+	at := gnmi.OC()
+	rib := at.NetworkInstance(ateDst.Name).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+	prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+		NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+
+	gnmi.WatchAll(t, ate, prefixPath.State(), time.Minute, func(v *ygnmi.Value[string]) bool {
+		_, present := v.Val()
+		return present
+	}).Await(t)
+
+	var wantASSeg = []uint32{dutAS, ateAS1}
+
+	if removeASPath {
+		for _, as := range asSeg {
+			if as < 64512 {
+				wantASSeg = append(wantASSeg, as)
+			}
+		}
+	} else {
+		wantASSeg = append(wantASSeg, asSeg...)
+	}
+
+	gotASSeg, ok := gnmi.WatchAll(t, ate, rib.AttrSetAny().AsSegmentAny().State(), 1*time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Protocol_Bgp_Rib_AttrSet_AsSegment]) bool {
```

Comment:
Looks like for this ATE Subscribe call, there are 2 lists returned for index 0 & 1, one for Public AS & Private AS. In that case the check cmp.Diff(val.Member, wantASSeg) == "" will never be true ?
e.g below are the values of val.Member lists retuned for case "AS Path SEQ - 65501, 65507, 65534” in this watch call.

AS List is [500 100]
AS List is [65501 65507 65534]

Reply:
ATE returns only one list with all the AS numbers before configuring "remove private as" and once "remove private as " is configured it will strip off private AS numbers and returns list with public AS numbers only.

Log captured from script:
===========================
bgp_remove_private_as_test.go:334: Verify AS Path list received at ate Port2 including private AS number.
    bgp_remove_private_as_test.go:265: val.Member is [500 100 65501 65507 65534]
    bgp_remove_private_as_test.go:266: wantASSeg s [500 100 65501 65507 65534]
    bgp_remove_private_as_test.go:337: Configure remove private AS on DUT.
    bgp_remove_private_as_test.go:340: Private AS numbers should be stripped off while advertising BGP routes into public AS.
    bgp_remove_private_as_test.go:265: val.Member is [500 100 65501 65507 65534]
    bgp_remove_private_as_test.go:266: wantASSeg s [500 100]
    bgp_remove_private_as_test.go:265: val.Member is [500 100 65501 65507 65534]
    bgp_remove_private_as_test.go:266: wantASSeg s [500 100]
|
|
|
|
 bgp_remove_private_as_test.go:265: val.Member is [500 100 600]
    bgp_remove_private_as_test.go:266: wantASSeg s [500 100 600]
  bgp_remove_private_as_test.go:343: 
        *** Stopping protocols to 10.92.196.126...
    bgp_remove_private_as_test.go:345: Remove remove-private-AS on DUT.

Reply:
Okay, its returning as two Lists for us from the ATE . [\"203.0.113.1\",\"32\",\"NA\",\"192.0.2.5\",\"removePacket[ ]\",\"removePacket[ ]\",\"0\",\"0\",\"IGP\",\"**<500 100> <65501 65507 65534>**. 
Could you share the IxOS version used in your tests ?

Reply:
I am using Ixia version is 9.30

Reply:
Thanks, we will check this with keysight team why there is a difference here.